### PR TITLE
Fixed minor issues 

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Fixes #
- Updated get posts query not to include my posts and added createdAt value in GSI1SK to sort it
- Refactored the accept donation lambda and checked if the user blood group matches the requested blood group
- Added HTTP logger for API gateway lambdas

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [x] Bugfix
- [x] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [ ] No risk
- [x] Low
- [ ] High

### CODE REVIEW

File: core/services/aws/bloodDonation/acceptDonationRequest.ts [hunk 1]
```diff
@@ -60,21 +67,42 @@ async function acceptDonationRequest(
...
-    if (acceptanceRecord !== null && acceptanceRecord.status === AcceptDonationStatus.COMPLETED) {
+    if (isAlreadyDonated(acceptanceRecord)) {
       throw new Error('You already donated.')
     }
 
-    if ((status === AcceptDonationStatus.ACCEPTED && acceptanceRecord !== null)) {
+    if (acceptanceRecord !== null && status === acceptanceRecord.status) {
       return generateApiGatewayResponse(
-        { message: 'Donation request accepted successfully.' },
+        { message: 'Donation request already accepted successfully.' },
```
Comment (sakibstark11) [inline core/services/aws/bloodDonation/acceptDonationRequest.ts:76]: `Donation request was accepted earlier`

Comment (sakibstark11) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts:76]: Maybe this is what you meant to say>?

File: core/services/aws/bloodDonation/acceptDonationRequest.ts [hunk 2]
```diff
@@ -60,21 +67,42 @@ async function acceptDonationRequest(
...
+          donationPost,
+          donorId,
+          createdAt,
+          status,
+          userProfile
+        )
+      }
     } else {
-      if (acceptanceRecord !== null) {
+      if (status === AcceptDonationStatus.IGNORED) {
```
Comment (sakibstark11) [inline core/services/aws/bloodDonation/acceptDonationRequest.ts:105]: This logic feels a little confusing,
you're saying that the acceptanceRecord can be null only when status === Accepted ?

File: core/services/aws/bloodDonation/acceptDonationRequest.ts [hunk 3]
```diff
@@ -167,12 +219,20 @@ async function acceptBloodDonationRequest(
...
     userId: seekerId,
-    title: 'Donor Found',
-    status: AcceptDonationStatus.ACCEPTED,
-    body: `${donationPost.requestedBloodGroup} blood found`,
+    title: status === AcceptDonationStatus.ACCEPTED ? 'Donor Found' : 'Donor Ignored',
+    status,
+    body:
+      status === AcceptDonationStatus.ACCEPTED
+        ? `${donationPost.requestedBloodGroup} blood found`
+        : 'unfortunately donor Ignored the request',
```
Comment (sakibstark11) [inline core/services/aws/bloodDonation/acceptDonationRequest.ts:227]: `request was ignored by donor`

File: core/services/aws/bloodDonation/acceptDonationRequest.ts (hunk 2 above)
Comment (mahinuzzaman) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts:105]: no, here the logic is
if acceptanceRecord !== null and then if status == ignore then we should delete the acceptance record

File: core/services/aws/bloodDonation/acceptDonationRequest.ts (hunk 3 above)
Comment (mahinuzzaman) [reply core/services/aws/bloodDonation/acceptDonationRequest.ts]: Updated in fcb443694a4c98a9ce4e50177bafa732c625dd75

**Pull request merged**